### ISSUE TITLE
sql: dropping a table prevents recreating a table with the same name

### ISSUE THREAD

**Issue by arjunravinarayan** [cockroachdb/cockroach#13700]
Consider the following set up commands, on a cluster that has a large existing table.
```
DROP TABLE usertable;
CREATE TABLE usertable(...);
> pq: table "usertable" is being dropped
```

This is problematic because:
1. The drop takes a lot of time for large tables. I'm at an hour and counting on a drop on cluster denim.
2. There is no status or progress report for how long this will take.
3. There is no way to "undo" this drop. For instance, I've got the advice "rename the table, then drop it" but once I start the drop it's too late. Now I have no choice but to wait (or wipe the cluster entirely and restart it).

I would like to propose that this is not an unusual situation, and one that people might encounter often (particularly when they are polishing a program that will create the table, so they want to recreate it after amending the `create table` portion of their program, rather than using an `alter table` directly). For instance, I've frequently done this exact sequence of table creation/dropping many times in a row in Rails when using Active Record while playing around with the right schema for my tables.

There's an easy fix here: when processing a `DROP TABLE foo`, we could automatically generate a random UUID, and do: `ALTER TABLE foo rename to uuid` followed by `DROP TABLE uuid`. This at least unblocks following commands.

I'm happy to take on this work, if there's consensus on this being the right solution. cc @petermattis, @cuongdo.

**Comment by bdarnell** [comment]
+1 on renaming the table before dropping (as long as it's done atomically, so that there is no chance of the table being left with a different name without starting the drop process). 

Bikeshedding a bit, I'd rename it to something like `$ORIG_NAME.drop.$UUID` so it's not quite so cryptic in `SHOW TABLES`.

**Comment by petermattis** [comment]
I'm fine with renaming and then dropping. +1 on Ben's suggestion to keep the original name as a prefix for the temporary name.

**Comment by vivekmenezes** [comment]
closing in favor of the same issue reported earlier #7348
